index.js

```diff
@@ -0,0 +1,161 @@
+const core = require('@actions/core');
+const exec = require('@actions/exec');
+
+const fs = require('fs');
+
+const { parse_bash_array, enable_plugins, move_all } = require('./util.js');
+
+// exec options
+const options = {
+  shell: '/bin/bash'
+};
+
+// Standard out and standard error will be written to these variables
+var myOutput = "";
+var myError = "";
+options.listeners = {
+  stdout: (data) => {
+    myOutput += data.toString();
+  },
+  stderr: (data) => {
+    myError += data.toString();
+  }
+};
+
+// Reading inputs
+const tutor_version = core.getInput('tutor_version');
+const tutor_pearson_plugin_url = core.getInput('tutor_pearson_plugin_url');
+const gh_access_token = core.getInput('gh_access_token');
+const tutor_pearson_plugins = core.getInput('tutor_pearson_plugin_name');
+const tutor_plugin_sources = core.getInput('tutor_plugin_sources');
+const tutor_plugin_names = core.getInput('tutor_plugin_names');
+const extra_private_requirements = core.getBooleanInput('extra_private_requirements');
+const private_repositories = core.getInput('private_repositories');
+const branches = core.getInput('branches');
```

Comment:
Variables will be ready for usage from here. @ShonTitor 
```suggestion
const tutor_pearson_plugins = parse_bash_arraycore.getInput('tutor_pearson_plugin_name'));
const tutor_plugin_sources = parse_bash_array(core.getInput('tutor_plugin_sources'));
const tutor_plugin_names = parse_bash_array(core.getInput('tutor_plugin_names'));
const extra_private_requirements = core.getBooleanInput('extra_private_requirements');
const private_repositories = parse_bash_array(core.getInput('private_repositories'));
const branches = parse_bash_array(core.getInput('branches'));
```